config/routes.rb

```diff
@@ -29,6 +29,13 @@
     omniauth_callbacks: "omniauth_callbacks",
   }
 
+  scope "/api", module: :api, defaults: {format: :json} do
+    get "/all_global_stats", to: "global_stats#all_global_stats"
+    get "/impressions_total_count", to: "global_stats#impressions_total_count"
+    get "/impressions_click_count", to: "global_stats#impressions_click_count"
+    get "/impressions_click_rate", to: "global_stats#impressions_click_rate"
+  end
+
```

Comment:
We need to keep our endpoints restful instead of RPC style. Something more along the lines of `global_stats#show` that returns everything. 

Also, I don't think our API surface area is large enough to warrant special distinction in a subdirectory yet.
I'd move this up as a first class citizen under `controllers` rather than under the `api` directory. _Consider that the API is simply an implementation detail of which templates render after the action method completes._

Reply:
Awesome. Refactored for the restful design like we talked about and removed the api namespace structure.